Make StatusBar Notification Icon Clickable

MikeTrahearn-Qinetic opened the pull request:
This PR to cover point 1) in #2072 which requests the Status Bar Notification Icon to be clickable and take you to the NotificationPage. Note, the use of Global.notificationLayer.popAndGoToNotifications() function was avoided as this is too heavy for this use case and causes the UI to refresh the NavBar in an undesirable manner. Instead, we just request to change the page, which invokes the correct swipe animation and keeps the NavBar intact.

Since the NotificationIcon is invisible when there are no notifications, the child TapHandler only works when there are any notifications.

Full Fixes #2072 completely as point 2 (noted by @mr-manuel) has been removed.

mr-manuel commented:
I removed point 2 for now.

chriadam reviewed (approved):
LGTM, tested fine on desktop and wasm.  I am not in the office today, so didn't test on Cerbo.

blammit commented on components/StatusBar.qml:
```diff
@@ -267,6 +267,10 @@ FocusScope {
 						"qrc:/images/icon_info_32.svg" : "qrc:/images/icon_warning_32.svg"
+
+			TapHandler {
+				onTapped: pageManager.navBar.setCurrentPage("NotificationsPage.qml")
```
Can you make this work with key navigation as well? (In that sense it might be easier to turn it into a `Button`, to get the space-to-click and navigation highlight for free.)

blammit replied:
Now that key nav has been disabled in `main`, this would be fine to merge as-is. @chriadam any thoughts?

MikeTrahearn-Qinetic replied:
@blammit if key navigation were added, it would simply be "disabled" because of #2161 until such time as it would be enabled again (or caught as part of further key navigation operational rework). If you think it would be wise to continue with the addition of "dormant" key navigation then I would value your assistance to understand the mechanism to adorn this button upgrade with this behaviour.

MikeTrahearn-Qinetic commented:
I pushed an update to bring the Notification Icon up to the same level of functionality of the other buttons:
- added support for KeyNavigation (tested by temporarily reverting Commit b4df1664 to re-enable it)
- fixed the layout location
- includes upgrade to StatusBarButton
- always shown like the clock, regardless if the opacity of the StatusBar
- clicking now shows the standard PressArea effect like other status bar buttons

Here is a preview of the button with the KeyNavigation also enabled:

<img width="912" alt="NotificationButton-Dark" src="https://github.com/user-attachments/assets/377f81dc-e53a-4fd6-a160-7552c5c6f74d" />
<img width="912" alt="NotificationButton-Light" src="https://github.com/user-attachments/assets/15e5763e-81c4-4b28-bb2d-df5b7e8a9829" />

blammit reviewed (commented):
Works nicely with the key navigation.

MikeTrahearn-Qinetic commented on components/StatusBar.qml:
```diff
@@ -254,20 +254,29 @@ FocusScope {
+			   ? Theme.color_critical
+			   : Global.notifications?.statusBarNotificationIconPriority === VenusOS.Notification_Warning
+				 ? Theme.color_warning :
+				   Global.notifications?.statusBarNotificationIconPriority === VenusOS.Notification_Info ? Theme.color_ok : notificationButton.color
```
Such a binding is legal - no warning would be shown. I can however change it to "transparent" since I believe in this case the icon wouldn't be displayed. This would at least remove the "apparent" possibility for a value lock, even though it never would actually occur.

blammit commented on components/StatusBar.qml:
```diff
@@ -254,20 +254,29 @@ FocusScope {
+			verticalCenter: parent.verticalCenter
 		}
+		// The notificationButton should always be shown, even when the StatusBar is not
+		opacity: 1
```
If the status bar is not shown, then the button will not be shown either. The button can only be seen when the status bar has an opacity > 0.
